Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -5563,6 +5585,13 @@ void WebPageProxy::didFinishDocumentLoadForFrame(FrameIdentifier frameID, uint64
     }
 }
 
+SubframePageProxy* WebPageProxy::subframePageProxyForFrameID(WebCore::FrameIdentifier identifier) const
+{
+    if (m_registrableDomainForSubframeID.contains(identifier))
```

Comment:
We're doing a double hash map lookup in m_registrableDomainForSubframeID here and on the next line. This is unnecessarily inefficient.